roofit/roofit/src/RooPolyFunc.cxx

```diff
@@ -0,0 +1,287 @@
+                            << ")" << endl;
+   }
+   int n_terms = _terms.size();
+   std::string coeff_name = Form("%s_c%d", GetName(), n_terms);
+   std::string term_name = Form("%s_t%d", GetName(), n_terms);
+   auto termList = std::make_unique<RooListProxy>(term_name.c_str(), term_name.c_str(), this);
+   auto coeff = new RooRealVar(coeff_name.c_str(), coeff_name.c_str(), coefficient);
+   termList->addOwned(exponents);
+   termList->addOwned(*coeff);
+   this->_terms.push_back(move(termList));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Default constructor
+
+RooPolyFunc::RooPolyFunc() {}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Parameterised constructor
+
+RooPolyFunc::RooPolyFunc(const char *name, const char *title, const RooAbsCollection &vars)
+   : RooAbsReal(name, title), _vars("x", "list of dependent variables", this)
+{
+   for (const auto &var : vars) {
+      if (!dynamic_cast<RooAbsReal *>(var)) {
+         std::stringstream ss;
+         ss << "RooPolyFunc::ctor(" << GetName() << ") ERROR: coefficient " << var->GetName()
+            << " is not of type RooAbsReal";
+         const std::string errorMsg = ss.str();
+         coutE(InputArguments) << errorMsg << std::endl;
+         throw std::runtime_error(errorMsg);
+      }
+      _vars.add(*var);
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Copy constructor
+
+RooPolyFunc::RooPolyFunc(const RooPolyFunc &other, const char *name)
+   : RooAbsReal(other, name), _vars("vars", this, other._vars)
+{
+   for (size_t i = 0; i < other._terms.size(); ++i) {
+      this->_terms.push_back(std::make_unique<RooListProxy>(other._terms[i]->GetName(), this, *(other._terms[i])));
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Evaluate value of Polynomial.
+double RooPolyFunc::evaluate() const
+{
+   // Calculate and return value of polynomial
+   double poly_sum(0.0);
+   for (const auto &term : _terms) {
+      double poly_term(1.0);
+      size_t n_vars = term->size() - 1;
+      for (size_t i_var = 0; i_var < n_vars; ++i_var) {
+         auto var = dynamic_cast<RooRealVar *>(_vars.at(i_var));
+         auto exp = dynamic_cast<RooRealVar *>(term->at(i_var));
+         poly_term *= pow(var->getVal(), exp->getVal());
+      }
+      auto coef = dynamic_cast<RooRealVar *>(term->at(n_vars));
+      poly_sum += coef->getVal() * poly_term;
+   }
+   return poly_sum;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Compute multiple values of Polynomial.
+// RooSpan<double> RooPolyFunc::evaluateSpan(RooBatchCompute::RunContext& evalData, const RooArgSet* normSet) const {
+// }
+
+////////////////////////////////////////////////////////////////////////////////
+/// Taylor expanding given function in terms of observables around
+/// observableValues.
+
```

Comment:
```suggestion
////////////////////////////////////////////////////////////////////////////////
/// Taylor expanding given function in terms of observables around
/// observableValues.
/// \param[in] func Some explanation ending with a dot.
/// \param[in] order Some explanation ending with a dot.
/// \param[in] observables Some explanation ending with a dot.
/// \param[in] observableValues Some explanation ending with a dot.
```

Please write some complete function documentation for this important function. In particular you should mention that only up to order 2 is supported.

(see for example https://github.com/root-project/root/blob/master/roofit/roofitcore/src/RooHelpers.cxx#L62 for examples of complete doxygen function docs).